### PR TITLE
refactor: Removed unused dependencies

### PR DESCRIPTION
**Pull request by frol**
In preparation for publishing nearcore crates, I discovered some unnecessary dependencies.

[cargo-udeps](https://github.com/est31/cargo-udeps) helped to indentify all the unused dependencies:

```
$ cargo +nightly udeps --workspace --all-targets --all-features --bins --tests --benches --examples
```

P.S. I had to use a newer rustc (a74d1862d 2020-05-14 worked for me)

UPD: I would like to integrate `cargo udeps` into CI, but that will require Rustc version bump (potentially happening in #2668) /cc @ailisp

### CODE REVIEW

File: core/primitives/Cargo.toml [hunk 1]
```diff
@@ -41,6 +38,9 @@ dump_errors_schema = ["near-rpc-error-macro/dump_errors_schema"]
 [dev-dependencies]
 bencher = "0.1.5"
 
+[package.metadata.cargo-udeps.ignore]
+normal = ["jemallocator"]
```
Comment (bowenwang1996) [inline core/primitives/Cargo.toml:42]: what's this for?

Comment (frol) [reply core/primitives/Cargo.toml:42]: It is to whitelist the dependency, so it is not marked as not used by cargo-udeps.

Comment (bowenwang1996) [reply core/primitives/Cargo.toml:42]: That I understand. What I meant is why you need to whitelist this dependency.

Comment (frol) [reply core/primitives/Cargo.toml:42]: cargo-udeps could not understand that it is used :man_shrugging:

**Pull request merged**